docs/proposals/initializers.md

```diff
@@ -0,0 +1,110 @@
+
+## Goals
+
+1. Enable asynchronous controller pattern (i.e. initializers) as a viable alternative to existing admission control to support loose coupling of components where synchronous user response is less critical.
+2. Enable registration of a set of initializers to an API resource that must be able to act on an object to support its initialization.
+
+## Model changes
+
+**Initializer**
+
+A new type introduced into the API that allows the registration of an initializer with an associated group and kind in the API.  This allows dynamic registration of initializers without a forced restart or recompilation of the server.
+
+**NOTE: It's possible we could support this without introduction of a new type, maybe via the ConfigData resource, needs thought and input from community**
+
+**ObjectMeta.Initializers**
+
+Map of named tokens representative of external controllers that must visit a resource prior to becoming active. The set of initializers are defaulted based on the API Group and Kind of resource as part of configuration using our existing admission control facility.  For example, if an Initializer (name=initial-sizer) is associated with (group/kind=pod), then when a pod is initially created via the API, it will have its ObjectMeta.Initializers field set to "initial-sizer".
+
+**ObjectMeta.Initialized**
+
+Boolean flag denoting if resource is or is not initialized. If a resource has active initializers, the value is false. Once all initializers have acted, the value is toggled to true. Once a resource is initialized, it is forever initialized. When a resource is not initialized, validation rules for editing are loosened to allow components to make delayed updates. An object must pass strict validation in order to become "initialized".
+
+## REST API Changes
+
+All API objects will support a new endpoint as a sub-resource that allows non-initialized objects to be edited via a PUT that does not carry the same validation constraints as normal updates of a resource.
+
+For example:
+
+```PUT /api/{version}/{group}/{kind}/{name}/initialize/{initializer-name}```
```

Comment:
It wouldn't trigger synchronous execution. The main point is to permit asynchronous, decoupled (out-of-binary) execution.